events/git-and-github.md

```diff
@@ -0,0 +1,126 @@
+# Git & Github Workshop
+
+## Aim of the Workshop
+
+At the end of the workshop, make sure that audience are comfortable with the following
+
+- Why Version Control Systems are need?
+- Usage of basic Git commands to track their projects - Adding and Committing
+- Be able to push and pull code from Github
+- Be able to send a PR to an Open Source Repository
+
+## Before the workshop
+
+Share an instructions file like [this](https://github.com/kossiitkgp/git-and-github-workshop-2020/blob/main/installation.md), along with the poster of the workshop. The instructions should contain
+
+- How to install Git for Windows, Linux, Mac seperately
+- For Windows users, make that they install `git bash`
+- How to verify that git is installed
+- Instructions on how to create a Github Account
```

Comment:
https://github.blog/2020-12-15-token-authentication-requirements-for-git-operations/  .so based on the new update GitHub CLI actions won't be authenticated so we need to include some more instructions if, like enable 2fa and setup pat or more simpler for workflow is ssh. so we need to tell some basics of ssh and install respective ssh clients . we need to initialize ssh key using some basic commands. we need to give a basic overview of ssh. I guess....

Reply:
Thanks a lot for pointing it out!
I almost missed this. 

So you say that adding SSH keys is the only simpler alternative. So I think these are things compulsory right
1. Having installed SSH
2. Having RSA keys

The best thing I feel is, we will ask them to install SSH, by mentioning instructions. And in the demo we will show how to generate keys and add them to Github. What do you think?

Also, a doubt - what about people using Git bash in windows. RSA keys can be generated on Git bash also naa?.

Reply:
Yeah we can generate keys on Windows too. It's stored in C:\Users\USERNAME\.ssh and has the same format and everything as the Linux ~/.ssh folder. Assuming OpenSSH is being used.

Reply:
https://docs.github.com/en/github/authenticating-to-github/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent#platform-windows 

Instructions for Windows

Reply:
Yes, but we ask the users to type commands in `git bash` rather than the windows terminal. So the git config set in `git bash` might not be same as in `windows terminal` right?

Reply:
It's stored in the same place (as far as I remember). So we can just run the commands on Git bash for generating SSH keys and everything will work.

Reply:
Thanks a lot @FadedCoder for the docs, it seems the instructions are the same for Git bash and Linux!

Reply:
> Thanks a lot for pointing it out!
> I almost missed this.
> 
> So you say that adding SSH keys is the only simpler alternative. So I think these are things compulsory right
> 
>     1. Having installed SSH
> 
>     2. Having RSA keys
> 
> 
> The best thing I feel is, we will ask them to install SSH, by mentioning instructions. And in the demo we will show how to generate keys and add them to Github. What do you think?
> 
> Also, a doubt - what about people using Git bash in windows. RSA keys can be generated on Git bash also naa?.


ha, update ##before the workshop--include ssh installation guides for all 3 OS.and we will ask them to generate keys in the workshop and add that to their respective accounts.

Reply:
Done
Added the changes!
Thank you everyone for the changes!